pkg/build/prune/prune.go

```diff
@@ -3,37 +3,58 @@ package prune
 import (
 	"time"
 
+	"github.com/golang/glog"
+
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
 )
 
-// PruneFunc is a function that is invoked for each item during Prune
-type PruneFunc func(build *buildapi.Build) error
+type Pruner interface {
+	// Prune is responsible for actual removal of builds identified as candidates
+	// for pruning based on pruning algorithm.
+	Prune(buildPruner BuildPruner) error
+}
 
-type PruneTasker interface {
-	// PruneTask is an object that knows how to execute a single iteration of a Prune
-	PruneTask() error
+// BuildPruner knows how to delete builds from OpenShift.
+type BuildPruner interface {
+	// PruneBuild deletes the build from OpenShift's storage.
+	PruneBuild(build *buildapi.Build) error
 }
 
-// pruneTask is an object that knows how to prune a data set
-type pruneTask struct {
+// pruner is an object that knows how to prune a data set
+type pruner struct {
 	resolver Resolver
-	handler  PruneFunc
 }
 
-// NewPruneTasker returns a PruneTasker over specified data using specified flags
-// keepYoungerThan will filter out all objects from prune data set that are younger than the specified time duration
-// orphans if true will include inactive orphan builds in candidate prune set
-// keepComplete is per BuildConfig how many of the most recent builds should be preserved
-// keepFailed is per BuildConfig how many of the most recent failed builds should be preserved
-func NewPruneTasker(buildConfigs []*buildapi.BuildConfig, builds []*buildapi.Build, keepYoungerThan time.Duration, orphans bool, keepComplete int, keepFailed int, handler PruneFunc) PruneTasker {
+var _ Pruner = &pruner{}
+
+// PrunerOptions contains the fields used to initialize a new Pruner.
+type PrunerOptions struct {
+	// KeepYoungerThan indicates the minimum age a BuildConfig must be to be a
```

Comment:
> I don't think the option is restrained just to build configs.

Ahh, you may disregard my previous comment. This is indeed restrained to build configs.